### PR TITLE
chore: Convert zoom-to-fit plugin to TypeScript.

### PR DESCRIPTION
**Pull request by gonfunko**
This PR fixes part of https://github.com/google/blockly-samples/issues/1647 by converting the zoom-to-fit plugin to Typescript.

### CODE REVIEW

File: plugins/zoom-to-fit/src/index.ts [hunk 1]
```diff
@@ -32,24 +32,24 @@ export class ZoomToFitControl implements Blockly.IPositionable {
   private top = 0;
   
   /**
-   * Width of the zoom-to-fit control.
+   * width of the zoom-to-fit control.
```
Comment (BeksOmega) [inline plugins/zoom-to-fit/src/index.ts]: Revert capitalization change, and in below docs for `height`

File: plugins/zoom-to-fit/tsconfig.json [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+    "target": "es6",
+    "allowJs": true,
+    "sourceMap": true,
+    "moduleResolution": "nodenext",
+    "lib": ["es2021", "dom"]
+  },
+  "include": [
+    "src", "test"
+  ],
+}
```
Comment (BeksOmega) [inline plugins/zoom-to-fit/tsconfig.json:13]: This does not match other tsconfigs in samples, (in particular the `target`, `module`, and `moduleResolution`). Not sure what it is supposed to look like though.

**Review by BeksOmega — COMMENTED** [review]
Generally LGTM! But not sure how the tsconfig is supposed to be set up.

File: plugins/zoom-to-fit/tsconfig.json (hunk 2 above)
Comment (gonfunko) [reply plugins/zoom-to-fit/tsconfig.json:13]: It's the same as what I used for toolbox-search, although I don't remember exactly how I arrived at that.

File: plugins/zoom-to-fit/src/index.ts (hunk 1 above)
Comment (gonfunko) [reply plugins/zoom-to-fit/src/index.ts]: Done!

File: plugins/zoom-to-fit/tsconfig.json (hunk 2 above)
Comment (BeksOmega) [reply plugins/zoom-to-fit/tsconfig.json:13]: Ok, could you file a bug for checking on this and updating the configs to be consistent?

**Review by BeksOmega — APPROVED** [review]

File: plugins/zoom-to-fit/tsconfig.json (hunk 2 above)
Comment (gonfunko) [reply plugins/zoom-to-fit/tsconfig.json:13]: Done, #1871.

**Pull request merged**